src/utils/IO/system_checks.jl

```diff
@@ -61,3 +61,19 @@ function critical_components_check(sys::System)
     missing_critical_components &&
     throw(IS.InvalidValue("Critical Componeents are not present."))
 end
+
+"""
+Checks the system for sum(generator ratings) >= sum(load ratings).
+
+# Arguments
+- `sys::System`: system
+"""
+function adequacy_check(sys::System)
+    total_gen = sum(get_rating.(get_components(Generator, sys))) * get_basepower(sys)
+    @debug "Total System Generation: $total_gen"
+    total_load =
+        sum(get_maxactivepower.(get_components(ElectricLoad, sys))) * get_basepower(sys)
+    @debug "Total System Load: $total_load"
+    total_load > total_gen &&
+    @error "System peak load ($total_load) exceeds total generation capability ($total_gen)."
```

Comment:
This should be a warning if someone intends to parse a system and then add components to it.

Reply:
That *should not* apply here.  `check!` should only be called on the system once it has been built, such as after parsing or deserialization from JSON.